vmm/src/lib.rs

```diff
@@ -1256,9 +1281,13 @@ impl Vmm {
 
         self.register_events()
             .map_err(|e| VmmActionError::StartMicrovm(ErrorKind::Internal, e))?;
-        self.start_vcpus(entry_addr)
+
+        let vcpus = self
+            .create_vcpus(entry_addr)
             .map_err(|e| VmmActionError::StartMicrovm(ErrorKind::Internal, e))?;
 
+        self.start_vcpus(vcpus)
```

Comment:
Is there anything missing here between calling `create_vcpus` and `start_vcpus`? It seems the IRQ chip still gets set up as part of `init_microvm` for all architectures, and `init_microvm` gets called before these two functions (line 1275).

Reply:
Creating the irqchip **after** calling `create_vcpus` is mandatory for aarch64 not x86_64 which is the reason behind carving out the `create_vcpus` logic from `start_vcpus`. No aarch64 logic was added yet. That will come in following PRs. So, all this PR does is move away `create_vcpu` logic for aarch64 specific purposes.

Reply:
We agreed to move to a separate function the setup of the interrupt controller for x86_64 to make more obvious the order of operations that need to be respected when initializing the microvm.

Reply:
Created `setup_interrupt_controller` function and did some additional refactoring.